### ISSUE TITLE
Control-z doesn't properly restore terminal features

### ISSUE THREAD

**Issue by asmeurer** [inducer/pudb#261]
Don't know if this should really be an urwid bug. If I control-z (suspend) in pudb, it prints the prompt below the PuDB UI. Inserting text works, but the cursor is not shown. On the other hand, when quitting PuDB, the UI is removed from the terminal history and the terminal state is properly restored.

**Comment by inducer** [comment]
Good point. I imagine we could just handle SIGTSTP by stopping Urwid and then resuming it on return.
